libjob_queue/src/torque_driver.c

```diff
@@ -490,19 +507,22 @@ static char* torque_driver_get_qstat_status(torque_driver_type * driver, char *
 job_status_type torque_driver_get_job_status(void * __driver, void * __job) {
   torque_driver_type * driver = torque_driver_safe_cast(__driver);
   torque_job_type * job = torque_job_safe_cast(__job);
+  int result = JOB_QUEUE_STATUS_FAILURE;
   char * status = torque_driver_get_qstat_status(driver, job->torque_jobnr_char);
-  int result = JOB_QUEUE_FAILED;
-  if (strcmp(status, "R") == 0) {
-    result = JOB_QUEUE_RUNNING;
-  } else if (strcmp(status, "E") == 0) {
-    result = JOB_QUEUE_DONE;
-  } else if (strcmp(status, "C") == 0) {
-    result = JOB_QUEUE_DONE;
-  } else if (strcmp(status, "Q") == 0) {
-    result = JOB_QUEUE_PENDING;
-  } else {
-    fprintf(stderr, "%s: Unknown status found (%s), expecting one of R, E, C and Q.\n", __func__, status);
-    result = JOB_QUEUE_STATUS_FAILURE;
+
+  if (status) {
```

Comment:
I would prefer to have early exit here ... `if (!status) { print; result = failure; free; return; } else ...`

But that's not a blocker, so I think we can just merge this.